pkg/proof/README.md

```diff
@@ -0,0 +1,198 @@
+This means that we will be proving that the row root commits to the share, and the data root commits to the row root.
+Since share *three* is committed to by `RowRoot1`, we will be proving that row root to the data root. 
+
+<img src="img/row_root_to_data_root_proof.png" alt="row root to data root proof" width="443"/>
+
+The inner nodes in maroon are the nodes needed to construct the merkle inclusion proof.
+
+So, if we have access to these two proofs,
+the [share to row root inclusion proof](#share-to-row-root-namespace-merkle-inclusion-proof) and [row root to data root inclusion proof](#row-root-to-data-root-inclusion-proof),
+we can successfully prove inclusion of share *three* to the Celestia data root.
+
+### Prove share commitment inclusion to data root
+
+To prove that a blob,
+referenced by its *share commitment*, was included in a Celestia block, we need the following proofs:
+
+#### Subtree roots inclusion proof to the data root
+
+As we've seen in the [row roots generation](#row-roots) section,
+the subtree roots are inner nodes of the namespace merkle tree used to generate the row root.
+This means that it's possible to create an inclusion proof of a subtree root to the row root:
+
+<img src="img/subtree_root_to_row_root.png" alt="subtree root to row root proof" width="443"/>
+
+The inner nodes in brown would be used to construct the `SR2` node inclusion proof to the `RowRoot3`.
+
+After having this proof, we will need to prove the `RowRoot3` inclusion to the data root.
+This is described in the [row root to data root inclusion proof](#row-root-to-data-root-inclusion-proof) section.
+
+With this, we would be able to prove that `SR2` was committed to by a Celestia data root.
+
+#### Subtree roots inclusion proof to the *share commitment*
+
+Now that we can prove `SR2` inclusion to `RowRoot3`,
+what's left is proving that `SR2` was committed to by the *share commitment*.
+
+Proving inclusion of `SR1` or `SR2` to the *share commitment* goes back
+to creating a binary merkle inclusion proof to the *share commitment*:
+
+<img src="img/share_commitment.png" alt="share commitment generation from subtree roots" width="300"/>
+
+So, if we manage to prove that `SR1` and `SR2` were both committed to by the Celestia data root, and that the *share commitment* was generated using `SR1` and `SR2`, then, we would have proven that the *share commitment* was committed to by the Celestia data root, which means that **the blob data that generated the *share commitment* was included in a Celestia block**.
+
+[IMPORTANT] As of the current version,
+the API for generating and verifying share commitment proofs to data root is still not exposed.
+These proofs are only used as part of the Celestia [consensus](https://celestiaorg.github.io/celestia-app/specs/consensus.html) internally.
+
+#### Compact proofs
+
+More compact proofs can be generated, but are out of the scope of this document.
```

Comment:
[optional]

throwing in a one sentence summary could be good here. like "Something something we can just prove the tx instead of the blob something something honest half quorum assumtion"

Reply:
[75c9054](https://github.com/celestiaorg/celestia-app/pull/3245/commits/75c90547fee1895afc95eca12540fce21d388b6b)